### ISSUE TITLE
Doing major changes in a published module can break Nuxt due to caching

### ISSUE THREAD

**Issue by rchl** [nuxt/nuxt.js#7838]
### Versions

- nuxt: v2.13.3
- node: v12

### Reproduction

On major changes to npm dependencies (seen with Nuxt modules at least), Nuxt can break in a way that will make it error out on start.

### Steps to reproduce

1. Have Nuxt running happily with the current version of modules.
2. Update some Nuxt module dependency that has made major changes to its structure.

### What is Expected?

Nuxt to run

### What is actually happening?

Getting an error like:

```
 FATAL  Template src not found: /Users/me/project/node_modules/@nuxtjs/sentry/lib/sentry.client.js

  at ModuleContainer.addTemplate (node_modules/@nuxt/core/dist/core.js:72:13)
  at ModuleContainer.addPlugin (node_modules/@nuxt/core/dist/core.js:92:26)
  at ModuleContainer.SentryModule (node_modules/@nuxtjs/sentry/lib/module.js:102:10)
  at ModuleContainer.addModule (node_modules/@nuxt/core/dist/core.js:229:34)
  at node_modules/@nuxt/utils/dist/utils.js:1821:43
  at async ModuleContainer.ready (node_modules/@nuxt/core/dist/core.js:49:5)
  at async Nuxt._init (node_modules/@nuxt/core/dist/core.js:699:5)


   ╭───────────────────────────────────────────────────────────────────────────────────────────────────────╮
   │                                                                                                       │
   │   ✖ Nuxt Fatal Error                                                                                  │
   │                                                                                                       │
   │   Error: Template src not found: /Users/me/project/node_modules/@nuxtjs/sentry/lib/sentry.client.js   │
   │                                                                                                       │
   ╰───────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

**Comment by rchl** [comment]
This issue is related to the `esm` caching.

Running `ESM_DISABLE_CACHE=true nuxt dev` fixes the issue (and makes the stale cache go away).

Note: I've noticed that `esm` caches code alongside the package, within `node_modules/[package]/node_modules/.cache/esm` so there is no central place where its cache is stored.

@pimlie @pi0 @clarkdo

**Comment by rchl** [comment]
I've reproduced this issue with `npm`. See https://github.com/rchl/nuxt-esm-cache-bug for steps.

Interestingly, I couldn't reproduce when using `yarn`...

**Comment by rchl** [comment]
It's the esm cache in `node_modules/@nuxt/core/node_modules/.cache/esm` that breaks this case.

**Comment by pi0** [comment]
Should be fixed with [v2.14.1](https://github.com/nuxt/nuxt.js/releases/tag/v2.14.1)

**Comment by rchl** [comment]
Thanks @clarkdo 👍
